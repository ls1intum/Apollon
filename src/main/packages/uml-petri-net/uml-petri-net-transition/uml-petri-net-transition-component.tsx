import React, { SFC } from 'react';
import { Text } from '../../../components/controls/text/text';
import { UMLPetriNetTransition } from './uml-petri-net-transition';

export const UMLPetriNetTransitionComponent: SFC<Props> = ({ element }) => (
  <g>
    <Text y="-15" fill={element.textColor}>
      {element.name}
    </Text>
    <rect
      width={element.bounds.width}
      height={element.bounds.height}
      stroke={element.strokeColor || 'black'}
      strokeWidth={2}
      fillOpacity={1}
    />
  </g>
);

interface Props {
  element: UMLPetriNetTransition;
}
