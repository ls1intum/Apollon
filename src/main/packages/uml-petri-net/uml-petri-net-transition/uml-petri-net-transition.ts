import { PetriNetElementType } from '..';
import { ILayer } from '../../../services/layouter/layer';
import { ILayoutable } from '../../../services/layouter/layoutable';
import { IUMLElement, UMLElement } from '../../../services/uml-element/uml-element';
import { UMLElementFeatures } from '../../../services/uml-element/uml-element-features';
import { IBoundary } from '../../../utils/geometry/boundary';
import { UMLElementType } from '../../uml-element-type';
import { DeepPartial } from '../../../typings';

export class UMLPetriNetTransition extends UMLElement {
  static features: UMLElementFeatures = { ...UMLElement.features };
  static defaultWidth = 20;
  static defaultHeight = 60;

  type: UMLElementType = PetriNetElementType.PetriNetTransition;
  bounds: IBoundary = {
    ...this.bounds,
  };

  constructor(values?: DeepPartial<IUMLElement>) {
    super(values);
    this.bounds.height = (values && values.bounds && values.bounds.height) || UMLPetriNetTransition.defaultHeight;
    this.bounds.width = (values && values.bounds && values.bounds.width) || UMLPetriNetTransition.defaultWidth;
  }

  render(layer: ILayer): ILayoutable[] {
    this.bounds.height = Math.max(this.bounds.height, UMLPetriNetTransition.defaultHeight);
    this.bounds.width = Math.max(this.bounds.width, UMLPetriNetTransition.defaultWidth);
    return [this];
  }
}
