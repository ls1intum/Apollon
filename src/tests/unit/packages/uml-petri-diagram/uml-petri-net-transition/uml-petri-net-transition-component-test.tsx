import React from 'react';
import { wrappedRender } from '../../../test-utils/render';
import { UMLPetriNetTransition } from '../../../../../main/packages/uml-petri-net/uml-petri-net-transition/uml-petri-net-transition';
import { UMLPetriNetTransitionComponent } from '../../../../../main/packages/uml-petri-net/uml-petri-net-transition/uml-petri-net-transition-component';

it('render the uml-petri-net-transition-component', () => {
  const element: UMLPetriNetTransition = new UMLPetriNetTransition({ name: 'Example Description' });
  const { getByText, baseElement } = wrappedRender(
    <svg>
      <UMLPetriNetTransitionComponent element={element} scale={1.0} />
    </svg>,
  );
  expect(getByText(element.name)).toBeInTheDocument();
  expect(baseElement).toMatchSnapshot();
});
