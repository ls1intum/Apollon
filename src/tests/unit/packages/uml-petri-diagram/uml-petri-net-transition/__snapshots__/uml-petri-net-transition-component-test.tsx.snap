// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the uml-petri-net-transition-component 1`] = `
<body>
  <div>
    <svg>
      <g>
        <text
          dominant-baseline="middle"
          font-weight="bold"
          pointer-events="none"
          text-anchor="middle"
          x="50%"
          y="-15"
        >
          Example Description
        </text>
        <rect
          class="sc-hLBbgP kcUggK"
          fill="white"
          fill-opacity="1"
          height="60"
          stroke="black"
          stroke-width="1"
          width="20"
        />
      </g>
    </svg>
  </div>
</body>
`;
